### PR TITLE
Add progress option for showing progress only

### LINKED ISSUE #1311: Show more user-friendly output, from the `start' command
### Description

Currently, the `limactl start` defaults to dumping the log with the INFO level to the console.

Turning off the logging means that only the final message (if any), was showing to the user.

I was experimenting with a less verbose mode, that only showed the major progress steps...

There were only two events in the start, Booting and Running, so I wanted to add requirements.

`Starting "default"`
`READY.`

Counted the total, and added a progress bar:
```
Starting "default"
Waiting 1 / 8 [--->________________________]  "ssh" (essential 1/5)             
Waiting 2 / 8 [------>_____________________]  "user session is ready for ssh" (essential 2/5)
Waiting 3 / 8 [---------->_________________]  "sshfs binary to be installed" (essential 3/5)
Waiting 4 / 8 [------------->______________]  "/etc/fuse.conf (/etc/fuse3.conf) to contain \"user_allow_other\"" (essential 4/5)
Waiting 5 / 8 [----------------->__________]  "the guest agent to be running" (essential 5/5)
Waiting 6 / 8 [-------------------->_______]  "systemd must be available" (optional 1/2)
Waiting 7 / 8 [------------------------>___]  "containerd binaries to be installed" (optional 2/2)
Waiting 8 / 8 [----------------------------]  "boot scripts must have finished" (final 1/1)
READY.
```

Found it less distracting, if the old steps remained.

### PR DESCRIPTION
**Pull request by afbjorklund**
After hiding the INFO messages, enable _some_ progress.

`Starting "default"`

`READY.`

Showing the requirements as a progress bar, using `pb`.

Closes #1311

### CODE REVIEW

**Comment by afbjorklund** [comment]
The first commit is from #1308, it messes up the progress bar to have both that and the log at the same time.

```
INFO[0105] [hostagent] Waiting for the essential requirement 1 of 5: "ssh" 
Waiting 1 / 8 [--->________________________]  "ssh" (essential 1/5)             INFO[0115] [hostagent] Waiting for the essential requirement 1 of 5: "ssh" 
Waiting 1 / 8 [--->________________________]  "ssh" (essential 1/5)             INFO[0132] [hostagent] Waiting for the essential requirement 1 of 5: "ssh" 
Waiting 1 / 8 [--->________________________]  "ssh" (essential 1/5)             INFO[0133] [hostagent] The essential requirement 1 of 5 is satisfied 
Waiting 1 / 8 [--->________________________]  "ssh" (essential 1/5)             
INFO[0133] [hostagent] Waiting for the essential requirement 2 of 5: "user session is ready for ssh" 
INFO[0133] [hostagent] The essential requirement 2 of 5 is satisfied 
Waiting 2 / 8 [------>_____________________]  "user session is ready for ssh" (essential 2/5)
```

**Comment by afbjorklund** [comment]
Fixed the width, when not keeping the output but cleaning the progress bar.

Haven't yet fully decided, which option is the best - there are pros and cons...

File: cmd/limactl/start.go [hunk 1]
```diff
@@ -61,6 +61,7 @@ $ cat template.yaml | limactl start --name=local -
 	startCommand.Flags().Bool("tty", isatty.IsTerminal(os.Stdout.Fd()), "enable TUI interactions such as opening an editor, defaults to true when stdout is a terminal")
 	startCommand.Flags().String("name", "", "override the instance name")
 	startCommand.Flags().String("set", "", "modify the template inplace, using yq syntax")
+	startCommand.Flags().Bool("progress", false, "show progress")
```
Comment (AkihiroSuda) [inline cmd/limactl/start.go:64]: This is confusing, because the progress shown in the default mode

Comment (afbjorklund) [reply cmd/limactl/start.go:94]: Needs to be split

**Pull request closed without merging**